crates/compiler/gen_llvm/src/llvm/expect.rs

```diff
@@ -564,6 +573,58 @@ fn load_tag_data<'a, 'ctx, 'env>(
     env.builder.new_build_load(tag_type, data_ptr, "load_data")
 }
 
+fn clone_tag_payload_and_id<'a, 'ctx, 'env>(
+    env: &Env<'a, 'ctx, 'env>,
+    layout_interner: &mut STLayoutInterner<'a>,
+    layout_ids: &mut LayoutIds<'a>,
+    ptr: PointerValue<'ctx>,
+    cursors: Cursors<'ctx>,
+    roc_union: RocUnion<'ctx>,
+    tag_id: usize,
+    payload_in_layout: InLayout<'a>,
+    opaque_payload_ptr: PointerValue<'ctx>,
+) -> IntValue<'ctx> {
+    let payload_type = basic_type_from_layout(env, layout_interner, payload_in_layout);
+
+    let payload_ptr = env.builder.build_pointer_cast(
+        opaque_payload_ptr,
+        payload_type.ptr_type(AddressSpace::default()),
+        "cast_payload_ptr",
+    );
+
+    let payload = env
+        .builder
+        .new_build_load(payload_type, payload_ptr, "payload");
+
+    // NOTE: `answer` includes any extra_offset that the tag payload may have needed
+    // (e.g. because it includes a list). That is what we want to return, but not what
+    // we need to write the padding and offset of this tag
+    let answer = build_clone(
+        env,
+        layout_interner,
+        layout_ids,
+        ptr,
+        cursors,
+        payload,
+        payload_in_layout,
+    );
+
+    // include padding between data and tag id
+    let tag_id_internal_offset = layout_interner.stack_size(payload_in_layout)
+        + roc_union.data_width()
+        - layout_interner.stack_size(payload_in_layout);
```

Comment:
this can be reduced to
```suggestion
    let tag_id_internal_offset = roc_union.data_width();
```

no?